tests/test_generate_files.py

```diff
@@ -291,3 +291,21 @@ def test_raise_undefined_variable_project_dir(tmpdir):
     assert error.context == {}
 
     assert not output_dir.join('testproject').exists()
+
+
+@pytest.mark.usefixtures('clean_system', 'remove_additional_folders')
+def test_external_templates():
+    os.mkdir('tests/custom_output_dir')
+    project_dir = generate.generate_files(
+        context={
+            'cookiecutter': {'food': 'pizzä'}
+        },
+        repo_dir=os.path.abspath('tests/test-generate-external-template'),
+        output_dir='tests/custom_output_dir',
+        extra_templates=os.path.abspath(
+            'tests/test-generate-external-template'
+        )
+    )
+    assert project_dir == os.path.abspath(
+        'tests/custom_output_dir/inputpizzä/'
+    )
```

Comment:
Some thoughts on this test:
- I don't think we need a custom output dir (and if we do, we should be using the `tmpdir` fixture to create it)
- I think we should be verifying the contents of the `extra-template.txt` file rather than the project_dir